cocotb/clock.py

```diff
@@ -65,12 +65,46 @@ class Clock(BaseClock):
         signal: The clock pin/signal to be driven.
         period (int): The clock period. Must convert to an even number of
             timesteps.
-        units (str, optional): One of 
+        units (str, optional): One of
             ``None``, ``'fs'``, ``'ps'``, ``'ns'``, ``'us'``, ``'ms'``, ``'sec'``.
             When no *units* is given (``None``) the timestep is determined by
             the simulator.
+
+
+    If you need more features like a phase shift and an asymmetric duty cycle,
+    it is simple to create your own clock generator (that you then :func:`fork`):
+
+    .. code-block:: python
+
+        @cocotb.coroutine
+        def custom_clock():
+            # pre-construct triggers for performance
+            high_time = Timer(high_delay)
+            low_time = Timer(low_delay)
+            yield Timer(initial_delay)
+            while True:
+                dut.clk <= 1
+                yield high_time
+                dut.clk <= 0
+                yield low_time
+
+    If you also want to change the timing during simulation,
+    use this slightly more inefficient example instead where 
+    the Timers inside the while loop are created with 
+    current delay values:
+
+    .. code-block:: python
+
+        @cocotb.coroutine
+        def custom_clock():
+            yield Timer(initial_delay)
+            while True:
+                dut.clk <= 1
+                yield Timer(high_delay)
+                dut.clk <= 0
+                yield Timer(low_delay)
```

Comment:
I still think this is probably hard to follow. Perhaps give the larger example:
```
        @cocotb.coroutine
        def custom_clock():
            while True:
                dut.clk <= 1
                yield Timer(high_delay)
                dut.clk <= 0
                yield Timer(low_delay)

        high_delay = low_delay = 100
        cocotb.fork(custom_clock())
        yield Timer(1000)
        high_delay = low_delay = 10  # change the clock speed
        yield Timer(1000)
```
Note that I dropped initial_delay, because changing that at runtime doesn't make sense.

Reply:
Alright, next version. :)